src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/rewriting/rules/RuleChangeOrderOfSubPlansOfMultiwayJoin.java

```diff
@@ -0,0 +1,55 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.rules;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.NaryLogicalOp;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.RuleApplication;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RuleChangeOrderOfSubPlansOfMultiwayJoin extends AbstractRewritingRuleImpl{
+
+    public RuleChangeOrderOfSubPlansOfMultiwayJoin( final double priority ) {
+        super(priority);
+    }
+
+    @Override
+    protected boolean canBeAppliedTo( final PhysicalPlan plan ) {
+        final PhysicalOperator op = plan.getRootOperator();
+        if ( IdentifyLogicalOp.matchMultiwayJoin(op) ) {
+            for ( int i = 0; i < plan.numberOfSubPlans(); i ++ ) {
+                final PhysicalOperator subOp = plan.getSubPlan(i).getRootOperator();
+                if ( IdentifyLogicalOp.matchJoin(subOp) ) {
+                    return true;
+                }
+            }
```

Comment:
What is this rule doing? According to its name, it changes the order of the subplans that are children of a multiway join. If that's the case, why do you need to check what type of root operator these subplans have?

Reply:
This rule does not change the order of sub-plans directly, the order of sub-plans of a multiway join is changed by rewriting the sub-plans (with a binary join as root). In another word, applying this rewriting rule will expand the child with binary join to two children of the multiway join.
Perhaps, I should rename this rewriting rule as RuleExpandBinaryJoinWithinMultiwayJoin. 
Does this name better explain the purpose of this rule? @hartig

Reply:
So, this rule removes a child join of a multiway join by merging it directly into the multiway join? In this case, call the class RuleMegeChildJoinIntoMultiwayJoin.